File: FestivalJava/Services/src/main/java/ModelServices/Interfaces/IServiceTransaction.java
package ModelServices.Interfaces;

import Domain.Transaction;
import Validation.Exceptions.ValidatorException;

import java.util.List;

/**
 * Created by Sebi on 28-Mar-17.
 */
public interface IServiceTransaction {
    /*
    Saves a transaction
     */
    void save(Transaction transaction) throws ValidatorException;

    /*
    Saves a transaction and generates id
     */
    void saveWithoutId(Transaction transaction) throws ValidatorException;

    /*
    Deletes a transaction
     */
    void delete(Integer idTransaction);

    /*
    Updates a transaction
     */
    void update(Integer idTransaction, Transaction transaction) throws ValidatorException;

    /*
    Gets a transaction by id
     */
    Transaction getById(Integer idTransaction);

    /*
    Gets all transactions
     */
    List<Transaction> getAll();
}

File: FestivalC#/Festival/Services/Service/ModelServices/IServiceTransaction.cs
using Festival.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Service
{
    public interface IServiceTransaction
    {
        /*
         * Saves a transaction
         */
        void save(Transaction transaction);

        /*
         * Deletes a transaction
         */
        void delete(int idTransaction);

        /*
         * Updates a transaction
         */
        void update(int id, Transaction transaction);

        /*
         * Gets a transaction
         */
        Transaction getById(int idArtist);

        /*
         * Returns all transactions
         */
        List<Transaction> getAll();

        /*
         * Saves a transaction without id
         */
        void saveWithoutId(Transaction transaction);
    }
}
